# RAT 6.3  

We will use the data in `https://s3-us-west-2.amazonaws.com/dsci/6002/data/homer.csv`.  

1) Fit a multiple regression model to predict $y$.  

In [1]:
%pylab inline

import pandas as pd
import seaborn as sns
import statsmodels.api as sm

Populating the interactive namespace from numpy and matplotlib


In [2]:
df = pd.read_csv("https://s3-us-west-2.amazonaws.com/dsci/6002/data/homer.csv")

In [3]:
df.head()

y        x1        x2        x3        x4        x5       x6
0 -2.17360 -0.023816 -0.012885  0.045936 -0.182890  0.028694 -0.81119
1 -1.29570  0.018453  0.058999 -0.072399  0.008679 -0.084344 -0.24208
2  0.14539  0.054999  0.131770 -0.067223  0.202430  0.127060  0.63388
3  0.92959 -0.018933 -0.010875 -0.028255  0.017912  0.084330  0.66062
4  0.90259 -0.000877  0.066941 -0.048190  0.064742  0.147880  0.26643

In [4]:
df.describe()

//anaconda/lib/python3.5/site-packages/numpy/lib/function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


y           x1           x2           x3           x4  \
count  6054.000000  6057.000000  6057.000000  6057.000000  6057.000000   
mean     -0.008547     0.000079    -0.000007    -0.000049     0.000068   
std       1.202365     0.039200     0.080779     0.058512     0.156589   
min      -3.810000    -1.000000    -0.303830    -0.199880    -0.633240   
25%            NaN    -0.021888    -0.054068    -0.039083    -0.109580   
50%            NaN    -0.000125     0.000111     0.000156    -0.000083   
75%            NaN     0.022555     0.055193     0.038948     0.107030   
max       4.035900     1.170000     0.292970     0.199290     0.533080   

                x5           x6  
count  6057.000000  6057.000000  
mean     -0.000027    -0.001662  
std       0.119576     0.784465  
min      -0.479130    -2.903200  
25%      -0.081445    -0.535640  
50%      -0.000788    -0.018469  
75%       0.079695     0.514390  
max       0.454490     3.619300

In [5]:
df = df.dropna(axis=0)

In [6]:
X = df[['x1', 'x2', 'x3', 'x4', 'x5','x6']]
X = sm.add_constant(X)
y = df['y']

model = sm.OLS(y, X)
results = model.fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.310
Model:                            OLS   Adj. R-squared:                  0.309
Method:                 Least Squares   F-statistic:                     452.1
Date:                Thu, 06 Oct 2016   Prob (F-statistic):               0.00
Time:                        13:59:00   Log-Likelihood:                -8583.7
No. Observations:                6054   AIC:                         1.718e+04
Df Residuals:                    6047   BIC:                         1.723e+04
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [95.0% Conf. Int.]
------------------------------------------------------------------------------
const         -0.0069      0.013     -0.534      0.593        -0.032     0.018
x1             4.0686      0.508      8.015      0.000         3.073     5.064
x2             1.1412      0.257      4.442      0.000         0.638     1.645
x3             4.0318      0.358     11.249      0.000         3.329     4.734
x4             0.9373      0.127      7.362      0.000         0.688     1.187
x5             3.9840      0.171     23.316      0.000         3.649     4.319
x6             0.9967      0.022     44.744      0.000         0.953     1.040
==============================================================================
Omnibus:                      635.831   Durbin-Watson:                   1.999
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              184.770
Skew:                           0.070   Prob(JB):                     7.54e-41
Kurtosis:                       2.156   Cond. No.                         41.2
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

2) Comment on the fit.  

The model explain around 30% of the variation in y.

3) Interpret the slope coefficient of $x_1$.  

For every unit of increase of x1, y increases by 0.508.

4) Is the model useful for predicting $y$? Why or why not?

F-stat: 452.1.

The overall p-value is 0.

We have enough evidence to conclude that at least one independent variable is significant.